src/unix/lwt_unix_unix.c

```diff
@@ -1156,10 +1153,28 @@ struct job_open {
 static void worker_open(struct job_open *job)
 {
   int fd;
+  int cloexec;
+
+  if (job->fd & CLOEXEC)
+    cloexec = 1;
+  else if (job->fd & KEEPEXEC)
+    cloexec = 0;
+  else
+#if OCAML_VERSION_MAJOR >= 4 && OCAML_VERSION_MINOR >= 5
+    cloexec = unix_cloexec_default;
+#else
+    cloexec = 0;
+#endif
+
+#if !defined(NEED_CLOEXEC_EMULATION)
```

Comment:
As I understand it, with this diff, `NEED_CLOEXEC_EMULATION` will no longer ever be defined.